File: AdSurgeDemo/AdSurgeDemo/AdSurgeDemoUI/AppOpenViewController.m
//
//  AppOpenViewController.m
//  AdSurgeDemo
//
//  Created by shirao on 2025/12/18.
//

#import <Foundation/Foundation.h>
#import <AdSurgeSDK/AdSurgeAppOpenAd.h>

#import "AdLogTable.h"
#import "AppOpenViewController.h"
#import "SettingManager.h"

@interface AppOpenViewController () <AdSurgeAppOpenAdDelegate>
@property (nonatomic, strong) AdSurgeAppOpenAd *appOpenAd;
@property (nonatomic, strong) AdLogTable *logTable;
@end

@implementation AppOpenViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.unitIds = @[@"16762", @"16763"];
    self.unitIdTitles = @[@"General 16762", @"Bidding 16763"];
    self.unitId = self.unitIds.firstObject;
    self.logTable = [[AdLogTable alloc] initWithParentView:self.view withTopMargin:20];
}

- (void)loadAd {
    self.appOpenAd = [[AdSurgeAppOpenAd alloc] initWithAdUnitIdentifier:self.unitId];
    self.appOpenAd.delegate = self;

    AdSurgeAdConfig *adConfig = [AdSurgeAdConfig configWithAdFormat:AdSurgeAdFormat.appOpen];
    adConfig.mute = [SettingManager sharedManager].isMuted;

    [self.appOpenAd loadAd:adConfig];
}

- (void)setNavigationBar {
    [super setNavigationBar];
    UILabel *label = [[UILabel alloc] init];
    label.text = @"AppOpenAD";
    label.textColor = [UIColor whiteColor];
    label.textAlignment = NSTextAlignmentCenter;
    label.font = [UIFont boldSystemFontOfSize:18];
    [label sizeToFit];
    self.navigationItem.titleView = label;
}

- (void)showBtnTapped:(UIButton *)sender {
    [super showBtnTapped:sender];
    if (![self.appOpenAd isValid]) {
        [self.logTable addLogWithInfo:@"Ad not valid" ad:nil error:nil reward:nil];
        return;
    }
    [self.appOpenAd showAdFromRootViewController:self];
}

#pragma mark - AdSurgeAppOpenAdDelegate

- (void)didLoadAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's format: %@, adUnitIdentifier: %@, revenue: %f, creative_id: %@",ad.format.label, ad.adUnitIdentifier, ad.revenue, ad.creativeId);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(AdSurgeError *)error {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's adUnitIdentifier: %@, error: %@",adUnitIdentifier, error);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:nil error:error reward:nil];
}

- (void)didDisplayAd:(AdSurgeAd *)ad withError:(AdSurgeError *)error {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"%@ \n",error);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:error reward:nil];
}

- (void)didHideAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didClickAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didPayRevenueForAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

@end

File: AdSurgeDemo-Swift/AdSurgeDemo-Swift/AdSurgeDemoUI/AppOpenViewController.swift
//
//  AppOpenViewController.swift
//  AdSurgeDemo-Swift
//
//  Created by shirao on 2026/1/8.
//

import UIKit
import AdSurgeSDK

class AppOpenViewController: BaseAdViewController, AdSurgeAppOpenAdDelegate {

    var appOpenAd: AdSurgeAppOpenAd!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.unitIds = ["16762", "16763"]
        self.unitIdTitles = ["General 16762", "Bidding 16763"]
        self.unitId = self.unitIds.first
    }

    override func setNavigationBar() {
        super.setNavigationBar()
        let label = UILabel()
        label.text = "AppOpenAD"
        label.textColor = .white
        label.textAlignment = .center
        label.font = UIFont.boldSystemFont(ofSize: 18)
        label.sizeToFit()
        self.navigationItem.titleView = label
    }

    override func loadAd() {
        guard let unitId = self.unitId else { return }
        self.appOpenAd = AdSurgeAppOpenAd(adUnitIdentifier: unitId)
        self.appOpenAd.delegate = self
        let adConfig = AdSurgeAdConfig(adFormat: .appOpen)
        self.appOpenAd.load(adConfig)
    }

    override func showBtnTapped(_ sender: UIButton) {
        super.showBtnTapped(sender)
        if (!self.appOpenAd.isValid) {
            addCell("Ad not valid")
            return
        }
        self.appOpenAd.show(fromRootViewController: self)
    }

    // MARK: - AdSurgeAppOpenAdDelegate

    func didLoad(_ ad: AdSurgeAd) {
        print(#function)
        print("ad's format: \(ad.format.label), adUnitIdentifier: \(ad.adUnitIdentifier), revenue: \(ad.revenue), creative_id: \(ad.creativeId)")
        addCell(#function)
    }

    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: AdSurgeError) {
        print(#function)
        print("ad's adUnitIdentifier: \(adUnitIdentifier), error: \(error)")
        addCell(#function)
    }

    func didDisplay(_ ad: AdSurgeAd, withError error: AdSurgeError?) {
        print(#function)
        print("\(String(describing: error))")
        addCell(#function)
    }

    func didHide(_ ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }

    func didClick(_ ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }

    func didPayRevenue(for ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }
}
